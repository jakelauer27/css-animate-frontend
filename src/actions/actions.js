export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadPrebuiltAnimations = (prebuiltAnimations) => ({
  type: 'LOAD_PREBUILT_ANIMATIONS',
  prebuiltAnimations
})

export const updateCurrentAnimation = (animation) => ({
  type: 'UPDATE_CURRENT_ANIMATION',
  animation
})

export const saveOriginalAnimation = (animation) => ({
  type: 'SAVE_ORIGINAL_ANIMATION',
  animation
})

export const loadMyAnimations = (myAnimations) => ({
  type: 'LOAD_MY_ANIMATIONS',
  myAnimations
})

export const deleteMyAnimation = (id) => ({
  type: 'DELETE_MY_ANIMATION',
  id
})

export const selectAnimationForEdit = (animation) => ({
  type: 'SELECT_ANIMATION_FOR_EDIT',
  animation
})

export const removeAnimationForEdit = () => ({
  type: 'REMOVE_ANIMATION_FOR_EDIT'
})


