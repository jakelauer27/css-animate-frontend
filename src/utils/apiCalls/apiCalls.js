export const addUser = async (user) => {
  try {
    const response = await fetch('https://css-animate.herokuapp.com/api/users/new', {
      method: 'POST',
      body: JSON.stringify(user),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    const data = await response.json()
    return data 
  } catch(error) {
    return new Error(error)
  }
}

export const addAnimation = async (user_id, ani_name, properties, keyframes) => {
  try {
    const response = await fetch(`https://css-animate.herokuapp.com/api/users/${user_id}/animations/new`, {
      method: 'POST',
      body: JSON.stringify({user_id, ani_name, properties, keyframes}),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    const data = await response.json()
    return data 
  } catch(error) {
    return new Error(error)
  }
}

export const editAnimation = async (user_id, animation_id, animation) => {
  try {
    const response = await fetch(`https://css-animate.herokuapp.com/api/users/${user_id}/animations/${animation_id}`, {
      method: 'PUT',
      body: JSON.stringify({user_id, animation_id, ani_name: animation.properties.name, properties: animation.properties, keyframes: animation.keyframes}),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    const data = response.json()
    return data
  } catch(error) {
    return new Error(error)
  }
}