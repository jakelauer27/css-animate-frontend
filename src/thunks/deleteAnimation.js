import { deleteMyAnimation } from '../actions/actions'

export const deleteAnimation = (user_id, animation_id) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`https://css-animate-backend.herokuapp.com/api/users/${user_id}/animations/${animation_id}`, {
        method: 'DELETE',
      })
      const data = await response.json()
      dispatch(deleteMyAnimation(animation_id))
      return data
    } catch (error) {
      return new Error(error)
    }
  } 
}