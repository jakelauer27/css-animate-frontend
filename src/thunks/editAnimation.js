import { editMyAnimation } from '../actions/actions'

export const editAnimation = (user_id, animation_id, animation) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`http://localhost:3000/api/users/${user_id}/animations/${animation_id}`, {
        method: 'PUT',
        body: JSON.stringify({user_id, animation_id, ani_name: animation.name, properties: animation.properties, keyframes: animation.keyframes}),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      const data = response.json()
      return data
    } catch(error) {
      return new Error(error)
    }
  }
}