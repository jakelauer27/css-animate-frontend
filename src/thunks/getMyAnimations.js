import { loadMyAnimations } from '../actions/actions'

export const getMyAnimations = (user_id) => {
  return async (dispatch) => {
    try {
      const response = await fetch(`http://localhost:3000/api/users/${user_id}/animations`)
      const data = await response.json()
      dispatch(loadMyAnimations(data.data))
    } catch(error) {
      return new Error(error)
    }
  }
}