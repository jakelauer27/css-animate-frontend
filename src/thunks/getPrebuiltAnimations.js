import { loadPrebuiltAnimations } from '../actions/actions'

export const getPrebuiltAnimations = () => {
  return async (dispatch) => {
    try {
      const response = await fetch("https://css-animate-backend.herokuapp.com/api/users/animations")
      const data = await response.json()
      dispatch(loadPrebuiltAnimations(data.data))
    } catch(error) {
      return new Error(error)
    }
  }
}