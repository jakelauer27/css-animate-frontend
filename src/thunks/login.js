import { signIn } from '../actions/actions'

export const loginUser = (user) => {
  return async (dispatch) => {
    try {
      const response = await fetch("https://css-animate-backend.herokuapp.com/api/users/", {
        method: 'POST',
        body: JSON.stringify(user),
        headers: {
          'Content-Type': 'application/json'
        }
      })
      const data = await response.json()
      dispatch(signIn(data.data))
      return data
    } catch(error) {
      return new Error(error)
    }
  }
}