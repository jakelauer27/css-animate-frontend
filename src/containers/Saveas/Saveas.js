import React, { Component } from 'react'
import * as API from '../../utils/apiCalls/apiCalls'
import { connect } from 'react-redux'
import { updateCurrentAnimation, saveOriginalAnimation } from '../../actions/actions'
import { getMyAnimations } from '../../thunks/getMyAnimations'
import { Redirect } from 'react-router-dom'

export class Saveas extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      redirect: false
    }
  }

  handleChange = (e) => {
    this.setState({name: e.target.value})
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    const { user_id, currentAnimation, updateCurrentAnimation, saveOriginalAnimation, getMyAnimations } = this.props
    let newAnimation = {...currentAnimation}
    newAnimation.properties.name = this.state.name
    newAnimation.keyframes.name = this.state.name
    newAnimation.name = this.state.name
    await API.addAnimation(user_id, this.state.name, JSON.stringify(newAnimation.properties), JSON.stringify(newAnimation.keyframes))
    await getMyAnimations(user_id)
    saveOriginalAnimation(newAnimation)
    updateCurrentAnimation(newAnimation)
    this.setState({redirect: true})
  }

  render() {
    if (this.state.redirect) {
      return <Redirect to='/properties'/>
    } 
    return (
      <form className='saveas-form'>
        <input type='text' value={this.state.name} placeholder='name' onChange={this.handleChange}>
        </input>
        <button onClick={this.handleSubmit} className='save-form-btn'>Save</button>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  user_id: state.user.id,
  currentAnimation: state.currentAnimation
})

export const mapDispatchToProps = (dispatch) => ({
  updateCurrentAnimation: (animation) => dispatch(updateCurrentAnimation(animation)),
  saveOriginalAnimation: (animation) => dispatch(saveOriginalAnimation(animation)),
  getMyAnimations: (id) => dispatch(getMyAnimations(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(Saveas)