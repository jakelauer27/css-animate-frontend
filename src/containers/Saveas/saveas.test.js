import { Saveas, mapStateToProps, mapDispatchToProps } from './Saveas'
import { shallow } from 'enzyme'
import React from 'react'
import { updateCurrentAnimation, saveOriginalAnimation } from '../../actions/actions'
import { getMyAnimations } from '../../thunks/getMyAnimations'
import * as API from '../../utils/apiCalls/apiCalls'

jest.mock('../../utils/apiCalls/apiCalls')
jest.mock('../../thunks/getMyAnimations')

API.addAnimation.mockImplementation(() => ({id: 1}))

describe('Saveas', () => {
  let mockClosePopup 
  let mockUpdateCurrentAnimation
  let mockSaveOriginalAnimation
  let mockGetMyAnimations
  let wrapper 
  const mockCurrentAnimation = {
    ani_name: 'slideInY',
    id: 1,
    user_id: 1,
    'properties': {
      'name': 'slideInY',
      'duration': '1.5s',
      'timingFunction': 'ease',
      'delay': '0s',
      'iterationCount': '1',
      'direction': 'normal',
      'fillMode': 'forwards'
    },
    'keyframes': {
      'name': 'slideInY',
      'sections': [
        {
          'name': '0%',
          'label': '0%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(-300px)'
            }
          ]
        },
        {
          'name': '100%',
          'label': '100%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(0px)'
            }
          ]
        },
      ]
    }
  }

  beforeEach(() => {
    mockClosePopup = jest.fn()
    mockUpdateCurrentAnimation = jest.fn()
    mockSaveOriginalAnimation = jest.fn()
    mockGetMyAnimations = jest.fn()
    wrapper = shallow(
      <Saveas 
        closePopup={mockClosePopup}
        user_id={1} 
        currentAnimation={mockCurrentAnimation}
        updateCurrentAnimation={mockUpdateCurrentAnimation}
        saveOriginalAnimation={mockSaveOriginalAnimation}
        getMyAnimations={mockGetMyAnimations}  
      />)
  })


  it('should match the snapshot if there is a user logged in', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if there is no user logged in', () => {
    wrapper = shallow(
      <Saveas 
        closePopup={mockClosePopup}
        user_id={null} 
        currentAnimation={mockCurrentAnimation}
        updateCurrentAnimation={mockUpdateCurrentAnimation}
        saveOriginalAnimation={mockSaveOriginalAnimation}
        getMyAnimations={mockGetMyAnimations}  
      />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should run cancel when the cancel button is click and a user is logged in', () => {
    const mockEvent = {preventDefault: jest.fn()}
    const spy = jest.spyOn(wrapper.instance(), 'cancel')
    wrapper.find('.cancel-form-btn').simulate('click', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('should run cancel when the cancel button is click and no user is logged in', () => {
    wrapper = shallow(
      <Saveas 
        closePopup={mockClosePopup}
        user_id={null} 
        currentAnimation={mockCurrentAnimation}
        updateCurrentAnimation={mockUpdateCurrentAnimation}
        saveOriginalAnimation={mockSaveOriginalAnimation}
        getMyAnimations={getMyAnimations}  
      />)
    
    const mockEvent = {preventDefault: jest.fn()}
    const spy = jest.spyOn(wrapper.instance(), 'cancel')
    wrapper.find('.cancel-form-btn').simulate('click', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('should run handleSubmit when the save form button is clicked', () => {
    const mockEvent = {preventDefault: jest.fn()}
    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit')
    wrapper.find('.save-form-btn').simulate('click', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('should run handleChange when input is changed', () => {
    const mockEvent = {preventDefault: jest.fn()}
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    wrapper.find('input').simulate('change', mockEvent)

    expect(spy).toHaveBeenCalled()
  })

  it('should have a default state', () => {
    const expected = {name: ''}
    expect(wrapper.state()).toEqual(expected)
  })

  describe('handleChange', () => {

    it('should update name in state', () => {
      const mockEvent = {target: {value: 'jake'}}
      const expected = { name: 'jake'}
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state()).toEqual(expected)
    })
  })

  describe('handleSubmit', () => {
    const mockEvent = {preventDefault: jest.fn()}

    it('should call API.addAnimation with the correct params', async () => {
      wrapper.instance().setState({name: 'test'})
      const expected = {...mockCurrentAnimation}
      expected.properties.name = 'test'
      expected.keyframes.name = 'test'
      expected.ani_name = 'test'

      await wrapper.instance().handleSubmit(mockEvent)

      expect(API.addAnimation).toHaveBeenCalledWith(1, 'test', JSON.stringify(expected.properties), JSON.stringify(expected.keyframes))
    })

    it('should call getMyAnimations with the correct params', async () => {
      await wrapper.instance().handleSubmit(mockEvent)

      expect(mockGetMyAnimations).toHaveBeenCalledWith(1)
    })

    it('should call saveOriginalAnimation with the correct params', async () => {
      wrapper.instance().setState({name: 'test'})
      const expected = {...mockCurrentAnimation}
      expected.properties.name = 'test'
      expected.keyframes.name = 'test'
      expected.ani_name = 'test'

      await wrapper.instance().handleSubmit(mockEvent)

      expect(mockSaveOriginalAnimation).toHaveBeenCalledWith(JSON.stringify(expected))
    })

    it('should call updateCurrentAnimation with the correct params', async () => {
      wrapper.instance().setState({name: 'test'})
      const expected = {...mockCurrentAnimation}
      expected.properties.name = 'test'
      expected.keyframes.name = 'test'
      expected.ani_name = 'test'

      await wrapper.instance().handleSubmit(mockEvent)

      expect(mockUpdateCurrentAnimation).toHaveBeenCalledWith(expected)
    })

    it('should call closePopup', async () => {
      await wrapper.instance().handleSubmit(mockEvent)

      expect(mockClosePopup).toHaveBeenCalled()
    })
  })

  describe('cancel', () => {

    it('should called closePopup', () => {
      const mockEvent = {preventDefault: jest.fn()}
      wrapper.instance().cancel(mockEvent)

      expect(mockClosePopup).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps', () => {
    const mockState = {currentAnimation: {keyframes: {name: 'slideInX'}}, user: {id: 1}}
    const expected = {currentAnimation: {keyframes: {name: 'slideInX'}}, user_id: 1}
    const result = mapStateToProps(mockState)
    expect(result).toEqual(expected)
  })

  describe('mapDispatchToProps', () => {

    it('should return a props object with a updateCurrentAnimation property', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = updateCurrentAnimation(mockCurrentAnimation)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.updateCurrentAnimation(mockCurrentAnimation)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with a saveOriginalAnimation property', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = saveOriginalAnimation(mockCurrentAnimation)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.saveOriginalAnimation(mockCurrentAnimation)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with a getMyAnimations property', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = getMyAnimations(1)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.getMyAnimations(1)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })


})