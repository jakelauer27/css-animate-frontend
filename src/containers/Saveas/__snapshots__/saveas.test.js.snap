// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saveas should match the snapshot if there is a user logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Saveas
    closePopup={[MockFunction]}
    currentAnimation={
      Object {
        "ani_name": "slideInY",
        "id": 1,
        "keyframes": Object {
          "name": "slideInY",
          "sections": Array [
            Object {
              "label": "0%",
              "name": "0%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(-300px)",
                },
              ],
            },
            Object {
              "label": "100%",
              "name": "100%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(0px)",
                },
              ],
            },
          ],
        },
        "properties": Object {
          "delay": "0s",
          "direction": "normal",
          "duration": "1.5s",
          "fillMode": "forwards",
          "iterationCount": "1",
          "name": "slideInY",
          "timingFunction": "ease",
        },
        "user_id": 1,
      }
    }
    getMyAnimations={[MockFunction]}
    saveOriginalAnimation={[MockFunction]}
    updateCurrentAnimation={[MockFunction]}
    user_id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="infoPopup-overlay"
        />,
        <form
          className="saveas-form"
        >
          <h2>
            Save Animation As
          </h2>
          <input
            onChange={[Function]}
            placeholder="name"
            type="text"
            value=""
          />
          <button
            className="save-form-btn"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
          <button
            className="cancel-form-btn cancel-saveas"
            onClick={[Function]}
          >
            cancel
          </button>
        </form>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "infoPopup-overlay",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Save Animation As
            </h2>,
            <input
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />,
            <button
              className="save-form-btn"
              disabled={true}
              onClick={[Function]}
            >
              Save
            </button>,
            <button
              className="cancel-form-btn cancel-saveas"
              onClick={[Function]}
            >
              cancel
            </button>,
          ],
          "className": "saveas-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save Animation As",
            },
            "ref": null,
            "rendered": "Save Animation As",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save",
              "className": "save-form-btn",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "cancel",
              "className": "cancel-form-btn cancel-saveas",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "cancel",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="infoPopup-overlay"
          />,
          <form
            className="saveas-form"
          >
            <h2>
              Save Animation As
            </h2>
            <input
              onChange={[Function]}
              placeholder="name"
              type="text"
              value=""
            />
            <button
              className="save-form-btn"
              disabled={true}
              onClick={[Function]}
            >
              Save
            </button>
            <button
              className="cancel-form-btn cancel-saveas"
              onClick={[Function]}
            >
              cancel
            </button>
          </form>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "infoPopup-overlay",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Save Animation As
              </h2>,
              <input
                onChange={[Function]}
                placeholder="name"
                type="text"
                value=""
              />,
              <button
                className="save-form-btn"
                disabled={true}
                onClick={[Function]}
              >
                Save
              </button>,
              <button
                className="cancel-form-btn cancel-saveas"
                onClick={[Function]}
              >
                cancel
              </button>,
            ],
            "className": "saveas-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save Animation As",
              },
              "ref": null,
              "rendered": "Save Animation As",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save",
                "className": "save-form-btn",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "cancel",
                "className": "cancel-form-btn cancel-saveas",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "cancel",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Saveas should match the snapshot if there is no user logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Saveas
    closePopup={[MockFunction]}
    currentAnimation={
      Object {
        "ani_name": "slideInY",
        "id": 1,
        "keyframes": Object {
          "name": "slideInY",
          "sections": Array [
            Object {
              "label": "0%",
              "name": "0%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(-300px)",
                },
              ],
            },
            Object {
              "label": "100%",
              "name": "100%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(0px)",
                },
              ],
            },
          ],
        },
        "properties": Object {
          "delay": "0s",
          "direction": "normal",
          "duration": "1.5s",
          "fillMode": "forwards",
          "iterationCount": "1",
          "name": "slideInY",
          "timingFunction": "ease",
        },
        "user_id": 1,
      }
    }
    getMyAnimations={[MockFunction]}
    saveOriginalAnimation={[MockFunction]}
    updateCurrentAnimation={[MockFunction]}
    user_id={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="infoPopup-overlay"
        />,
        <form
          className="saveas-form"
        >
          <h2>
            Access this feature by logging in
          </h2>
          <button
            className="cancel-form-btn cancel-saveas"
            onClick={[Function]}
          >
            Ok
          </button>
        </form>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "infoPopup-overlay",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Access this feature by logging in
            </h2>,
            <button
              className="cancel-form-btn cancel-saveas"
              onClick={[Function]}
            >
              Ok
            </button>,
          ],
          "className": "saveas-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Access this feature by logging in",
            },
            "ref": null,
            "rendered": "Access this feature by logging in",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ok",
              "className": "cancel-form-btn cancel-saveas",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Ok",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="infoPopup-overlay"
          />,
          <form
            className="saveas-form"
          >
            <h2>
              Access this feature by logging in
            </h2>
            <button
              className="cancel-form-btn cancel-saveas"
              onClick={[Function]}
            >
              Ok
            </button>
          </form>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "infoPopup-overlay",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Access this feature by logging in
              </h2>,
              <button
                className="cancel-form-btn cancel-saveas"
                onClick={[Function]}
              >
                Ok
              </button>,
            ],
            "className": "saveas-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Access this feature by logging in",
              },
              "ref": null,
              "rendered": "Access this feature by logging in",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ok",
                "className": "cancel-form-btn cancel-saveas",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Ok",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
